### PR TITLE
[SofaDefaulttype] Add override keyword (and remove thousands of warnings)

### PR DESCRIPTION
**Pull request by fredroy**
Especially relevant for clang compilers.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/typeinfo/DataTypeInfoDynamicWrapper.h [hunk 1]
```diff
@@ -77,8 +77,8 @@ class DataTypeInfoDynamicWrapper : public AbstractTypeInfo, public Info
 
     static AbstractTypeInfo* get() { static DataTypeInfoDynamicWrapper<Info> t; return &t; }
 
-    virtual const TypeInfoId& getBaseTypeId() const { return TypeInfoId::GetTypeId<typename Info::BaseType>(); }
-    virtual const TypeInfoId& getValueTypeId() const { return TypeInfoId::GetTypeId<typename Info::ValueType>(); }
+    virtual const TypeInfoId& getBaseTypeId() const override { return TypeInfoId::GetTypeId<typename Info::BaseType>(); }
```
Comment (jnbrunet) [inline SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/typeinfo/DataTypeInfoDynamicWrapper.h]: Maybe remove the virtual keyword while being here? It is redundant if you put the override.

Comment (fredroy) [reply SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/typeinfo/DataTypeInfoDynamicWrapper.h]: can do

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
#1634 is slightly more complete, closing this one.

**Pull request closed without merging**